sql/src/main/java/io/crate/executor/transport/task/UpsertByIdTask.java

```diff
@@ -124,22 +123,17 @@ public void execute(final RowReceiver rowReceiver, Row parameters) {
                 return;
             }
         }
-        Futures.addCallback(result, new FutureCallback<Long>() {
-            @Override
-            public void onSuccess(@Nullable Long result) {
-                rowReceiver.setNextRow(new Row1(result));
+        result.whenComplete(FutureCompleteConsumer.build(
```

Comment:
Are you asking about the general use of FutureCompletConsumer? Or just this particular case? 
Generally speaking, the FutureCompletConsumer abstraction is there to make sure the success/failure callbacks are called when they need to (ie call success when the received throwable is null, and failure otherwise - computation which would need to be present in every ad-hoc biconsumer). I'm happy to discuss alternatives if there are better ideas around this. The added overhead of the FCC objects might be an issue in certain cases, and if that's one of those I'm happy to drop it.